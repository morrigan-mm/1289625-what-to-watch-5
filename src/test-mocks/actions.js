export const unknownAction = {type: `NOOP`};
