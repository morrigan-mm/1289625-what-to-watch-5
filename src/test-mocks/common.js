export const mockEvent = {
  preventDefault: () => {}
};

export const noop = () => {};
