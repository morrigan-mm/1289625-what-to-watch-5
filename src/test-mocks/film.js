const film = {
  id: 15,
  poster: `#poster`,
  preview: `#preview`,
  video: `#video`,
  cover: `#cover`,
  title: `The Revenant`,
  genre: `Action`,
  releaseDate: 2015,
  description: `A frontiersman on a fur trading`,
  rating: 4,
  ratingCount: 618498,
  director: `The Director`,
  actors: [`actor1`, `actor2`],
  runtime: 156,
  addedToMyList: false,
  backgroundColor: `#92918B`,
  backgroundImage: `#bgimage`
};

export default film;
