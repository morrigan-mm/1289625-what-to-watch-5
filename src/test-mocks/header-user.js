import {AuthorizationStatus} from "../constants";

export const headerUserAuthorized = {
  authorizationStatus: AuthorizationStatus.AUTH,
  avatar: `#`
};
export const headerUserUnauthorized = {
  authorizationStatus: AuthorizationStatus.NO_AUTH
};
