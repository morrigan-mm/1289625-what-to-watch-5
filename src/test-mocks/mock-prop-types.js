import PropTypes from "prop-types";

export const mockComponentType = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};
