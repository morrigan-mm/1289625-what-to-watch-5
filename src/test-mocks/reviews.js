const reviews = [
  {
    id: 1,
    text: `some review text 1`,
    rating: 1.1,
    author: `Author name1`,
    reviewDate: `2020-01-01T14:13:56.569Z`
  },
  {
    id: 2,
    text: `some review text 2`,
    rating: 2.2,
    author: `Author name2`,
    reviewDate: `2020-02-02T14:13:56.569Z`
  },
  {
    id: 3,
    text: `some review text 3`,
    rating: 3.3,
    author: `Author name3`,
    reviewDate: `2020-03-03T14:13:56.569Z`
  }
];

export default reviews;
