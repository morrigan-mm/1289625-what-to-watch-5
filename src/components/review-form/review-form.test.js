import React from "react";
import renderer from "react-test-renderer";
import {noop} from "../../test-mocks/common";
import ReviewForm from "./review-form";

describe(`ReviewForm`, () => {
  it(`Should render correctly`, () => {
    const tree = renderer
      .create(
          <ReviewForm onSubmit={noop} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render all disabled when disabled=true`, () => {
    const tree = renderer
      .create(
          <ReviewForm onSubmit={noop} disabled />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });


  it(`Should render error message when submit failed with error status 401`, () => {
    const tree = renderer
      .create(
          <ReviewForm onSubmit={noop} addReviewError={401} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render error message when submit failed with other error status`, () => {
    const tree = renderer
      .create(
          <ReviewForm onSubmit={noop} addReviewError={500} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
