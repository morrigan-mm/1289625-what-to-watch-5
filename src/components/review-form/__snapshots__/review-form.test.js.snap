// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm Should render all disabled when disabled=true 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={true}
        className="rating__input"
        disabled={true}
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="0"
      />
      <label
        className="visually-hidden rating__label"
        htmlFor="star-0"
      >
        Rating 
        0
      </label>
      <input
        checked={false}
        className="rating__input"
        disabled={true}
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        disabled={true}
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={false}
        className="rating__input"
        disabled={true}
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        disabled={true}
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        disabled={true}
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      disabled={true}
      id="review-text"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;

exports[`ReviewForm Should render correctly 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={true}
        className="rating__input"
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="0"
      />
      <label
        className="visually-hidden rating__label"
        htmlFor="star-0"
      >
        Rating 
        0
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      id="review-text"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;

exports[`ReviewForm Should render error message when submit failed with error status 401 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <p
    className="error-message"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Sorry, you have to be authorized for that!
  </p>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={true}
        className="rating__input"
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="0"
      />
      <label
        className="visually-hidden rating__label"
        htmlFor="star-0"
      >
        Rating 
        0
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      id="review-text"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;

exports[`ReviewForm Should render error message when submit failed with other error status 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <p
    className="error-message"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops! Something goes wrong... Please, try again.
  </p>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        checked={true}
        className="rating__input"
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="0"
      />
      <label
        className="visually-hidden rating__label"
        htmlFor="star-0"
      >
        Rating 
        0
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        checked={false}
        className="rating__input"
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      id="review-text"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
      value=""
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        disabled={true}
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
</form>
`;
